### DISCUSSION TITLE
Mutliple colors in a `lineChart()`

### DISCUSSION THREAD

**Discussion by suboyar** [patrykandpatrick/vico#259] [Ideas]
Hi. Is it possible to create a line chart that has different colors? I tried to create a list of colors that has the size of model and passed this list to `ProvideChartStyle()`, but had no success. What I want to achieve is to have different color if its over the threshold value,  or if its not it should use another color instead.
![image](https://github.com/patrykandpatrick/vico/assets/46026042/f5b2c93d-8f92-4ef2-8202-c8324b06e7cf)

**Comment by suboyar** [comment]
After some perusing it seems that this is not currently supported as its written in #188

**Comment by patrickmichalik** [comment]
Hello! We plan on making this possible—I’ll be reopening this discussion and moving it to the “Ideas” category.

**Reply by DannyBloky** [reply]
Hi, is this already in progress or is there a solution for this.

**Reply by Gowsky** [reply]
Hello! We plan on making this possible.

**Reply by DannyBloky** [reply]
Oke thanks, do you know when you are going to start with this and what the plan is to finish this. 
Because then I know when to come back to you when I have a question about it.

Maybe good to know what I want to have.
Is that I have a threshold or 2 thresholds

So I want the line above the threshold be a other color

And with two thresholds it must be above and under to be a other color

**Reply by Gowsky** [reply]
We’ll share an update once it’s done.

**Comment by Gowsky** [comment]
Hi @suleymanboyar, @DannyBloky, [Vico 2.0.0-alpha.26](https://patrykandpatrick.com/vico/releases/2.0.0-alpha.26/) makes this possible via `LineCartesianLayer.LineFill.double` and `splitY` parameter. Cheers!

**Reply by scimone** [reply]
This is great, thank you!
Is it also possible to set 2 threshold values? And then set the line under one threshold to one color, and above the other threshold to another color?

**Reply by Gowsky** [reply]
@scimone, yes, by creating a custom `LineFill` implementation, as mentioned in the documentation. We plan on adding a built-in implementation that supports this in the future.

**Comment by Gowsky** [comment]
[Vico 2.0.0](https://github.com/patrykandpatrick/vico/releases/tag/v2.0.0) is now available.
